geospatial/src/Dem.cc

```diff
@@ -285,25 +295,48 @@ bool Dem::GeoReference(double _x, double _y,
   double geoTransf[6];
   if (this->dataPtr->dataSet->GetGeoTransform(geoTransf) == CE_None)
   {
-    OGRSpatialReference sourceCs;
-    OGRSpatialReference targetCs;
     OGRCoordinateTransformation *cT;
     double xGeoDeg, yGeoDeg;
 
-    // Transform the terrain's coordinate system to WGS84
-    const char *importString
-        = strdup(this->dataPtr->dataSet->GetProjectionRef());
-    if (importString == nullptr || importString[0] == '\0')
+    if (this->dataPtr->sphericalCoordinates.Surface() ==
+        math::SphericalCoordinates::EARTH_WGS84)
     {
-      gzdbg << "Projection coordinate system undefined." << std::endl;
-      return false;
+      OGRSpatialReference sourceCs;
+      OGRSpatialReference targetCs;
+      // Transform the terrain's coordinate system to WGS84
+      const char *importString
+          = strdup(this->dataPtr->dataSet->GetProjectionRef());
+      if (importString == nullptr || importString[0] == '\0')
+      {
+        gzdbg << "Projection coordinate system undefined." << std::endl;
+        return false;
+      }
+      sourceCs.importFromWkt(&importString);
+      targetCs.SetWellKnownGeogCS("WGS84");
+      cT = OGRCreateCoordinateTransformation(&sourceCs, &targetCs);
+    }
+    else if (this->dataPtr->sphericalCoordinates.Surface() ==
+        math::SphericalCoordinates::MOON_SCS)
+    {
+      auto sourceCs = this->dataPtr->dataSet->GetSpatialRef();
+      OGRSpatialReference targetCs = OGRSpatialReference();
+
+      const char* MOON_LAT_LON_PROJ_STR =
+        "+proj=latlong +a=1737400 +b=1737400";
+
+      targetCs.importFromProj4(MOON_LAT_LON_PROJ_STR);
+
+      cT = OGRCreateCoordinateTransformation(
+          sourceCs, &targetCs);
     }
-    sourceCs.importFromWkt(&importString);
-    targetCs.SetWellKnownGeogCS("WGS84");
-    cT = OGRCreateCoordinateTransformation(&sourceCs, &targetCs);
+    else
+    {
+      gzerr << "Unsupported spherical coordinates found" << std::endl;
+    }
+
     if (nullptr == cT)
```

Comment:
there's a compiler warning that `cT` may be used uninitialized. I think it should be initialized to `nullptr` on line 298

Reply:
Done : https://github.com/gazebosim/gz-common/pull/377/commits/5bb4c5b9de1bd0f5a7be67de17c0cf3a8e71b5f1